src/main/java/org/jabref/logic/util/JavaVersionCheck.java

```diff
@@ -0,0 +1,67 @@
+package org.jabref.logic.util;
+
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+/**
+ * Provides simple checks to ensure the correct version for JabRef is available. Currently, we need to make sure that we
+ * have Java 1.8 but not Java 9. The functions here are not intended for direct use. Instead, they are called inside
+ * {@link BuildInfo}, which has the required java version string (e.g. 1.8.0_144) available through the build system.
+ * The version check should always happen through the <code>Globals#BUILD_INFO</code> instance which is available at a
+ * very early stage in the JabRef startup.
+ */
+public class JavaVersionCheck {
+
+    // We expect version in the form 1.8.0_101 and need . and _ as delimiter to separate numbers
+    private static final Pattern DELIMITER = Pattern.compile("[._]");
+    // Be adventurous and assume that we can access this property always!
+    private static final String JAVA_VERSION = System.getProperty("java.version");
+
+    private JavaVersionCheck() {
```

Comment:
I would propose to make this class non-static. You then have a (public) constructor that accepts the current version string and a default constructor which uses `JAVA_VERSION`. In this way you can also easily write a test to verify the methods in this class.

Reply:
Yes, good catch.